dataservice/extensions/flask_indexd.py

```diff
@@ -33,14 +39,28 @@ def new_session(self):
         s = requests.Session()
         s.auth = (current_app.config['INDEXD_USER'],
                   current_app.config['INDEXD_PASS'])
-        s.headers.update({'Content-Type': 'application/json'})
+        s.headers.update({'Content-Type': 'application/json',
+                          'User-Agent': 'Kids First Dataservice'})
         return s
 
     def teardown(self, exception):
         ctx = stack.top
         if hasattr(ctx, 'indexd_session'):
             ctx.indexd_session.close()
 
+    def prefetch(self, dids):
+        """
+        Fetch a list of documents by did into the page cache.
+        """
+        # If running in dev mode, don't call indexd
+        if self.url is None or self.bulk_url is None:
+            return
+        print(self.url, self.bulk_url)
+        self.page_cache = {}
```

Comment:
Clearing the cache here means that the looped calls to indexd.prefetch inside of indexd_pagination will step on each other. Is that a problem? It seems like it would be better to clear the cache at the beginning of indexd_pagination instead.

Reply:
Yeah, that does seem to be an issue. I feel a little uneasy about clearing the cache outside of the indexd client though.